astropy/nddata/nduncertainty.py

```diff
@@ -190,191 +276,483 @@ def parent_nddata(self):
+            # uncertainty to the same unit as it's parent
+            if self.unit != self.parent_nddata.unit:
+                left = ((self.array * self.unit).to(
+                        self.parent_nddata.unit).value *
+                        other_uncert.parent_nddata.data)
+            else:
+                left = self.array * other_uncert.parent_nddata.data
+
+            if other_uncert.unit != other_uncert.parent_nddata.unit:
+                right = ((other_uncert.array * other_uncert.unit).to(
+                        other_uncert.parent_nddata.unit).value *
+                        self.parent_nddata.data)
+            else:
+                right = other_uncert.array * self.parent_nddata.data
+
+            if correlation != 0:
+                corr = (2 * correlation * left * right)
+                return np.sqrt(left**2 + right**2 + corr)
+            else:
+                return np.sqrt(left**2 + right**2)
+
+    @format_doc(_propagate_doc, operation='divison', operator='/',
+                instance='StdDevUncertainty')
+    def _propagate_divide(self, other_uncert, result_data, correlation):
+
+        # For division we don't need the result as quantity
+        if isinstance(result_data, Quantity):
+            result_data = result_data.value
 
         if self.array is None:
-            raise ValueError("standard deviation values are not set")
+            # Formula sigma = |(A / B) * (dB / B)|
+
+            # We need (db / B) to be dimensionless so we convert (if necessary)
+            # dB to the same unit as B
+            if other_uncert.unit != other_uncert.parent_nddata.unit:
+                right = ((other_uncert.array * other_uncert.unit).to(
+                    other_uncert.parent_nddata.unit).value /
+                    other_uncert.parent_nddata.data)
+            else:
+                right = (other_uncert.array / other_uncert.parent_nddata.data)
+            return np.abs(result_data * right)
+
+        elif other_uncert.array is None:
+            # Formula sigma = dA / |B|.
 
-        if other_nddata.uncertainty.array is None:
-            raise ValueError("standard deviation values are not set "
-                             "in other_nddata")
+            # We need to convert dA to the unit of A to have a result that
+            # matches the resulting data's unit.
+            if self.unit != self.parent_nddata.unit:
+                left = (self.array * self.unit).to(
+                        self.parent_nddata.unit).value
+            else:
+                left = self.array
+            return np.abs(left / other_uncert.parent_nddata.data)
 
-        result_uncertainty = StdDevUncertainty()
-        result_uncertainty.array = \
-            (np.sqrt((self.array/self.parent_nddata.data)**2
-             + (other_nddata.uncertainty.array/other_nddata.data)**2) *
-             result_data)
+        else:
+            # Formula sigma = |AB|*sqrt((dA/A)**2+(dB/B)**2-2*dA/A*dB/B*cor)
```

Comment:
I think `|AB|` → `|A/B|`?

I should also add the comment with the error propagation formulae are VERY helpful!